### PR TITLE
Add fix for Download-Only deployment type on DDI-API

### PR DESCRIPTION
**Pull request by laverman**
For action type _Download Only_ it doesn't make sense to send `deployment.download = attempt, deployment.update = attempt` since the device shall only download the artefact not install it.

Therefore, the response should look like the following:
`deployment.download = forced, deployment.update = skip`

Since this feature is quite new, I hope breaking the API here is acceptable? However, I feel the current behaviour is not what one would expect.

Feedback welcome!

PS: I fixed some typos along the way

Signed-off-by: Jeroen Laverman <jeroen.laverman@bosch-si.com>

### CODE REVIEW

**Comment by laverman** [comment]
@hawkbit-bot please verify

File: hawkbit-rest/hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java [hunk 1]
```diff
@@ -313,6 +313,14 @@ private static boolean checkModule(final String fileName, final SoftwareModule m
         return ResponseEntity.notFound().build();
     }
 
+    private HandlingType calculateDownloadType(final Action action) {
+        if (action.isDownloadOnly() || action.isForce()) {
+            return HandlingType.FORCED;
+        } else {
```
Comment (a-sayyed) [inline hawkbit-rest/hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java]: else could be removed, or use the `? :` operator:
`return action.isDownloadOnly() || action.isForce() ? HandlingType.FORCED : HandlingType.ATTEMPT;`

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/Action.java [hunk 2]
```diff
@@ -142,6 +142,13 @@ default boolean isForced() {
         return ActionType.FORCED == getActionType();
     }
 
+    /**
+     * @return true when action is downloadonly, false otherwise
+     */
+    default boolean isDownloadOnly() {
```
Comment (a-sayyed) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/Action.java]: We need to check where in the code the comparison we call `action.getType() == DOWNLOAD_ONLY`  and replace it with this

File: hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java [hunk 3]
```diff
@@ -539,6 +540,133 @@ public void deplomentAutoForceAction() throws Exception {
...
+    @Description("Attempt/soft deployment to a controller. Checks if the resource response payload for a given deployment is as expected.")
+    public void deploymentDownloadOnlyAction () throws Exception {
+        // Prepare test data
+        final DistributionSet ds = testdataFactory.createDistributionSet("", true);
+        final DistributionSet ds2 = testdataFactory.createDistributionSet("2", true);
+        final String visibleMetadataOsKey = "metaDataVisible";
+        final String visibleMetadataOsValue = "withValue";
+
+        final int artifactSize = 5 * 1024;
+        final byte random[] = RandomUtils.nextBytes(artifactSize);
```
Comment (a-sayyed) [inline hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java]: replace with `final byte[] random`

File: hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java [hunk 4]
```diff
@@ -539,6 +540,133 @@ public void deplomentAutoForceAction() throws Exception {
...
+
+        List<Target> saved = deploymentManagement.assignDistributionSet(ds.getId(), ActionType.DOWNLOAD_ONLY,
+                RepositoryModelConstants.NO_FORCE_TIME, Collections.singletonList(savedTarget.getControllerId()))
+                .getAssignedEntity();
+        assertThat(deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())).hasSize(1);
+
+        final Action action = deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())
+                .getContent().get(0);
+        assertThat(deploymentManagement.countActionsAll()).isEqualTo(1);
+        saved = assignDistributionSet(ds2, saved).getAssignedEntity();
```
Comment (a-sayyed) [inline hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java]: assignment not used

File: hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java [hunk 5]
```diff
@@ -539,6 +540,133 @@ public void deplomentAutoForceAction() throws Exception {
         assertThat(actionStatusMessage.getStatus()).isEqualTo(Status.RETRIEVED);
     }
 
+    @Test
+    @Description("Attempt/soft deployment to a controller. Checks if the resource response payload for a given deployment is as expected.")
+    public void deploymentDownloadOnlyAction () throws Exception {
```
Comment (a-sayyed) [inline hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java]: this test is very long, maybe the test setup could be refactored into separate method(s)?

File: hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java [hunk 6]
```diff
@@ -539,6 +540,133 @@ public void deplomentAutoForceAction() throws Exception {
         assertThat(actionStatusMessage.getStatus()).isEqualTo(Status.RETRIEVED);
     }
 
+    @Test
+    @Description("Attempt/soft deployment to a controller. Checks if the resource response payload for a given deployment is as expected.")
```
Comment (laverman) [inline hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java]: Fix description

File: hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java (hunk 5 above)
Comment (laverman) [reply hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiDeploymentBaseTest.java]: Will do this in another PR

**Comment by a-sayyed** [comment]
looks good 👍

**Pull request merged**